### PR TITLE
added position holdings view for issuer of asset

### PR DESCRIPTION
**Pull request by Sarahb-da**
The one caveat is that observers are added when the deposit is made by the investor from the UI, so when bootstrapping data locally the issuer will not be an observer of the asset until the investor's wallet screen is rendered. 

In a demo setting, when manually creating the deposits this should not be an issue.. 

Screen Shot 2021-05-19 at 11 29 48 AM

### CODE REVIEW

File: ui/src/pages/custody/Account.tsx [hunk 1]
```diff
@@ -28,13 +30,52 @@ const AccountComponent: React.FC<RouteComponentProps & ServicePageProps<Service>
...
+    if (retries > 0) {
+      await halfSecondPromise();
+      return updateDeposits(retries - 1);
+    }
+
+    return Promise.all(
+      deposits.map(d => {
+        damlSetValues(d.payload.asset.id.signatories).map(signatory =>
+          console.log(d.payload.asset.id.label, signatory, !d.observers.includes(signatory))
+        );
```
Comment (brian-weir) [inline ui/src/pages/custody/Account.tsx:61]: Do you want to keep this logging ?

File: ui/src/pages/custody/Account.tsx [hunk 2]
```diff
@@ -28,13 +30,52 @@ const AccountComponent: React.FC<RouteComponentProps & ServicePageProps<Service>
...
+
+        const newObservers = damlSetValues(d.payload.asset.id.signatories).filter(
+          signatory => !d.observers.includes(signatory)
+        );
+        console.log(newObservers);
```
Comment (brian-weir) [inline ui/src/pages/custody/Account.tsx:66]: Same as above ^^

File: ui/src/pages/custody/Account.tsx (hunk 1 above)
Comment (Sarahb-da) [reply ui/src/pages/custody/Account.tsx:61]: woops, no  - thanks!!!

**Review by brian-weir — APPROVED** [review]
lgtm!! 👍

**Comment by alexgraham-da** [comment]
> The one caveat is that observers are added when the deposit is made by the investor from the UI, so when bootstrapping data locally the issuer will not be an observer of the asset until the investor's wallet screen is rendered.

I believe you could fix that bootstrapping issue by changing `depositAsset` in `daml/Common.daml` to this:

```haskell
depositAsset : Providers -> Customer -> Asset -> Id -> Script (ContractId AssetDeposit)
depositAsset Providers{..} Customer{..} asset accountId = do
  -- Assets
  creditAccountRequestCid <- submit customer do exerciseCmd custodyServiceCid Custody.RequestCreditAccount with accountId = accountId; asset
  depositCid <- submit bank do exerciseCmd custodyServiceCid Custody.CreditAccount with ..
  (Some deposit) <- queryContractId bank depositCid
  submit customer do exerciseCmd depositCid AssetDeposit_SetObservers with newObservers = deposit.asset.id.signatories `union` deposit.observers
```

**Review by alexgraham-da — APPROVED** [review]
looks good!

**Pull request merged**